woocommerce-nosto-tagging.php

```diff
@@ -645,27 +656,81 @@ public function add_page_bottom_elements() {
 		$default_element_ids = array(
 			'nosto-page-bottom',
 		);
-		$element_ids         = apply_filters( 'wcnt_add_page_bottom_elements', $default_element_ids );
-		if ( is_array( $element_ids ) && ! empty( $element_ids ) ) {
-			$this->render( 'nosto-elements', array( 'element_ids' => $element_ids ) );
+		$element_ids = apply_filters( 'wcnt_add_page_bottom_elements', $default_element_ids );
+		if ( is_array( $element_ids ) && !empty( $element_ids ) ) {
```

Comment:
In the overloaded method outlined in another comment, you could have the page-type as a default null value. In the same method you could do the `is_array( $element_ids ) && !empty( $element_ids )` check. This will reduce the amount of boilerplate.